src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -25,7 +25,7 @@ namespace Microsoft.Extensions.FileProviders.Physical
     /// </summary>
     public class PhysicalFilesWatcher : IDisposable
     {
-        private static readonly Action<object> _cancelTokenSource = state => ((CancellationTokenSource)state).Cancel();
+        private static readonly Action<object?> _cancelTokenSource = state => ((CancellationTokenSource?)state)?.Cancel();
```

Comment:
avoid any product changes here. Please do them separately (I'd suggest filing an issue, ideally with a repro if you think this is a product bug), otherwise `!` if you feel confident enough this will not be null

Reply:
I don't think, this would ever be `null` (`Action<object?>` is required by contract, but I think non-null state is always passed into it). I'll double check and remove `?` from here.